src/lib/Output/Generator/InMemory/Xml.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Rest\Output\Generator\InMemory;
+
+use Ibexa\Rest\Output\Generator\Json;
+use Symfony\Component\Serializer\Encoder\XmlEncoder;
+
+final class Xml extends Json
```

Comment:
Is this not a mistake? Shouldnt it extend `Ibexa\Rest\Output\Generator\Xml` instead?

Reply:
No, the whole concept of "new" Xml is that it relies on Json Generator to allow Symfony's encoder to encode the array object into 'xml'.

Reply:
Yeah, this needs A LOT of clarification :sweat_smile: 

And needs to be adjusted before going final. These need to be generic containers for data, unrelated to JSON nor XML.

We treat `Generators` as if they are containers for a response. They contain a part of the response relevant to current `Visitor` context.

When looking at `Json`, we realized it actually contained the whole response tree, and only moved "an internal pointer" to a different location.

In contrast, `Xml` tried to output the result immediately.

So, `Generator`s are, in practice, "return values" for `Visitor`s. `Json` just fits because it already does what we want - contains complete data structure for us, that can be used in `AdapterNormalizer` to bridge the difference between our `Visitor`s and Symfony's Normalizers.

Reply:
>>> Is this not a mistake? Shouldnt it extend `Ibexa\Rest\Output\Generator\Xml` instead?

>> No, the whole concept of "new" Xml is that it relies on Json Generator to allow Symfony's encoder to encode the array object into 'xml'.

> Yeah, this needs A LOT of clarification 😅
> 
> And needs to be adjusted before going final. These need to be generic containers for data, unrelated to JSON nor XML.
> 
> We treat `Generators` as if they are containers for a response. They contain a part of the response relevant to current `Visitor` context.
> 
> When looking at `Json`, we realized it actually contained the whole response tree, and only moved "an internal pointer" to a different location.
> 
> In contrast, `Xml` tried to output the result immediately.
> 
> So, `Generator`s are, in practice, "return values" for `Visitor`s. `Json` just fits because it already does what we want - contains complete data structure for us, that can be used in `AdapterNormalizer` to bridge the difference between our `Visitor`s and Symfony's Normalizers.

This is very confusing. If both generators share the same concepts can we separate one common abstract which will be the base for Json (unchanged) and Xml (with changes)?